outline/connectivity.go

```diff
@@ -68,12 +86,12 @@ func CheckTCPConnectivityWithHTTP(client shadowsocks.Client, targetURL string) e
 	if !hasPort(targetAddr) {
 		targetAddr = net.JoinHostPort(targetAddr, "80")
 	}
-	conn, err := client.DialTCP(nil, targetAddr)
+	conn, err := dialer.Dial(ctx, targetAddr)
 	if err != nil {
 		return &ReachabilityError{err}
 	}
 	defer conn.Close()
-	conn.SetDeadline(time.Now().Add(tcpTimeout))
+	conn.SetDeadline(deadline)
```

Comment:
no need to change, just wondering, `ctx` already contains deadline info, why we still need to `SetDeadline`.

Reply:
Because the connection is not aware of the context we created. The context passed to Dial is only about the Dial operation.